scripts/pylib/twister/twisterlib/runner.py

```diff
@@ -511,6 +546,71 @@ def process(self, pipeline, done, message, lock, results):
             else:
                 self.cleanup_artifacts()
 
+    # Both old and new ztest fx co-exist for now.
+    # - The amending logic removes false tests introduced by old regex-based scanning.
+    # - The new collection logic adds the ones missed by the old regex-based scanning.
+    # When the old ztest fx is fully deprecated, the new collecting logic can work alone.
+    def collect_and_amend_testcases(self, results):
+        symbol_file = os.path.join(self.build_dir, "zephyr", "zephyr.symbols")
+        if os.path.isfile(symbol_file):
+            logger.debug(f"zephyr.symbols found: {symbol_file}")
+        else:
+            # No zephyr.symbols file, cannot do symbol-based test case collection
+            logger.debug(f"zephyr.symbols NOT found: {symbol_file}")
+            return
+
+        yaml_testsuite_name = self.instance.testsuite.id
```

Comment:
I see the yaml as more of a test config than suite, could we rename this?

Reply:
Yes, yaml file is a test config file.
But unfortunately, it is represented with the `class TestSuite`:
https://github.com/zephyrproject-rtos/zephyr/blob/384ff96b255b03b92aee50d5d036cb78971950b8/scripts/pylib/twister/twisterlib/testsuite.py#L339

So I follow it to avoid more confusion in jargons.